src/metabase/integrations/slack.clj

```diff
@@ -135,55 +145,81 @@
   "Make a GET request to a Slack API list `endpoint`, returning a sequence of objects returned by the top level
   `results-key` in the response. If additional pages of results exist, fetches those lazily, up to a total of
   `max-list-results`."
-  [endpoint results-key params]
+  [endpoint response->data params]
   ;; use default limit (page size) of 1000 instead of 100 so we don't end up making a hundred API requests for orgs
   ;; with a huge number of channels or users.
   (let [default-params {:limit 1000}
-        response       (m/mapply GET endpoint (merge default-params params))]
+        response       (m/mapply GET endpoint (merge default-params params))
+        data           (response->data response)]
     (when (seq response)
       (take
        max-list-results
        (concat
-        (get response results-key)
+        data
         (when-let [next-cursor (next-cursor response)]
           (lazy-seq
-           (paged-list-request endpoint results-key (assoc params :cursor next-cursor)))))))))
+           (paged-list-request endpoint response->data (assoc params :cursor next-cursor)))))))))
 
 (defn conversations-list
-  "Calls Slack API `conversations.list` and returns list of available 'conversations' (channels and direct messages). By
-  default only fetches channels."
+  "Calls Slack API `conversations.list` and returns list of available 'conversations' (channels and direct messages).
+  By default only fetches channels, and returns them with their # prefix. Note the call to [[paged-list-request]] will
+  only fetch the first [[max-list-results]] items."
   [& {:as query-parameters}]
   (let [params (merge {:exclude_archived true, :types "public_channel"} query-parameters)]
-    (paged-list-request "conversations.list" :channels params)))
+    (paged-list-request "conversations.list"
+                        ;; response -> channel names
+                        #(->> %
+                              :channels
+                              (map (fn [channel]
+                                     (str \# (:name channel)))))
+                        params)))
 
-(defn channel-with-name
-  "Return a Slack channel with `channel-name` (as a map) if it exists."
+(defn channel-exists?
+  "Returns true if the channel it exists."
```

Comment:
This function was only being used for this purpose before, so I made that expilcit here.